src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -332,11 +333,13 @@ public synchronized void commit() throws IOException {
             if (forceSync) {
                 long startSyncNS = System.nanoTime();
 
-                log.getChannel().force(false);
+                FileChannel channel = log.getChannel();
+                channel.force(false);
 
                 syncElapsedMS = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startSyncNS);
                 if (syncElapsedMS > fsyncWarningThresholdMS) {
-                    LOG.warn("fsync-ing the write ahead log in "
+                    LOG.warn("fsync-ing the write ahead log ("
+                            + channel.size() + " bytes) in "
```

Comment:
@afine Thank you for code review. When we got the warn message, will confuse that reason is `too large log` or `too busy disk I/O`? So, need to show the size of log to eliminate the `too large log` possibility.